sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/rx/ReadFeedExceptionHandlingTest.java

```diff
@@ -46,7 +49,10 @@ public void readFeedException() throws Exception {
                                                                     .mergeWith(Flux.fromIterable(frps));
 
         final CosmosAsyncClientWrapper mockedClientWrapper = Mockito.spy(new CosmosAsyncClientWrapper(client));
-        Mockito.when(mockedClientWrapper.readAllDatabases()).thenReturn(UtilBridgeInternal.createCosmosPagedFlux(pagedFluxOptions -> response));
+        Mockito.when(mockedClientWrapper.readAllDatabases()).thenReturn(UtilBridgeInternal.createCosmosPagedFlux(pagedFluxOptions -> {
+            pagedFluxOptions.setTracerInformation(new TracerProvider(ServiceLoader.load(Tracer.class)), "testSpan", "testEndpoint,", "testDb");
+            return response;
+        }));
```

Comment:
what update is this related to?
We don't validate anything related to tracing here.

Reply:
This is mock readAllDatabases() api which does not set tracer information. With the new tracer code all the query should go through tracer code pipeline and this is mandatory. If we dont do this, it will return null pointer  in page flux [here ](https://github.com/simplynaveen20/azure-sdk-for-java/blob/9c1d31f8b46f43c312fbb3dfcecc3837369d5e08/sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/util/CosmosPagedFlux.java#L93).
We can avoid this by doing null check in page flux, but the problem comes if someone add new query api , there are chances they will forget to send the request via tracer pipeline, and we don't want this.

Reply:
Explained offline